### DISCUSSION TITLE
Integration to the LiveSplit tool

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#145] [Ideas]
Hello!

Recently, Eclipsed has requested the adding an ability to add the integration with the [LiveSplit](https://livesplit.org/) program that is used by speed-runners to record their run results. There is a suggestion to export the state of the in-game timer into the running LiveSplit thing.

Right now I am not sure how I can do that, is there is a protocol specification that I can use?

**Comment by Wohlstand** [comment]
Created an issue https://github.com/Wohlstand/TheXTech/issues/224 because there is some info was given by LiveSplit folks, now I understand how I can implement the support.

**Comment by 0lhi** [comment]
Compat Settings and Speedrun Settings have different needs and in the future will require different filetypes to enhance configurability.

In my opinion, during the next Pull Request we should transfer all [Speedrun Configs](https://www.speedrun.com/smbx/guide/1blsy) inside a ```speedrun.ini```. This ```speedrun.ini``` could be stored inside the episode root or inside the gamesave root. Some new values that it could get in the future are Categories that could be selected at the start-speedrun-menu:

```
[category]
name = "Any%"
stop-timer-by = event
stop-timer-at = "Final Boss Event"

[category]
name = "100%"
stop-timer-by = enter
stop-timer-at = SecretCredits.lvl
```

And eventually, it could be used for LiveSplit Integration as discussed in #224:

```
[category]
splits = "Any.lss"
```
